staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -280,6 +280,7 @@ func (e *Store) GetExportStrategy() rest.RESTExportStrategy {
 // List returns a list of items matching labels and field according to the
 // store's PredicateFunc.
 func (e *Store) List(ctx context.Context, options *metainternalversion.ListOptions) (runtime.Object, error) {
+	ctx, _ = context.WithTimeout(ctx, 10*time.Second)
```

Comment:
But why the time is 10s?

Reply:
It's an example of showing timeout context works. If this solution is accepted, I can change it to a command line flag or a more appropriate value.

Reply:
I think this timeout size setting needs to be supported by the basis, not an experimental value.
:)